community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/CommunityLockClient.java

```diff
@@ -270,21 +270,24 @@ private boolean reEnter( PrimitiveLongObjectMap<LockResource> localLocks, long r
     }
 
     @Override
-    public void releaseShared( ResourceType resourceType, long resourceId )
+    public void releaseShared( ResourceType resourceType, long... resourceIds )
     {
         stateHolder.incrementActiveClients( this );
         try
         {
             PrimitiveLongObjectMap<LockResource> localLocks = localShared( resourceType );
-            LockResource resource = localLocks.get( resourceId );
-            if ( resource.releaseReference() != 0 )
+            for ( long resourceId : resourceIds )
             {
-                return;
-            }
+                LockResource resource = localLocks.get( resourceId );
+                if ( resource.releaseReference() != 0 )
+                {
+                    return;
```

Comment:
Do we not want to try releasing any remaining resources after the first failure? I don't know, maybe this will terminate the transaction and release everything anyway, just noting this observation.